tensorflow/python/ops/custom_gradient.py

```diff
@@ -406,14 +406,17 @@ def internal_grad_fn(unused_op, *result_grads):  # pylint: disable=unused-variab
 
 def _eager_mode_decorator(f, args, kwargs):
   """Implement custom gradient decorator for eager mode."""
-  with tape_lib.VariableWatcher() as variable_watcher:
-    result, grad_fn = f(*args, **kwargs)
+
+  trainable_vars = []
+  if 'trainable_variables' in kwargs:
```

Comment:
This is an implicit API change where you now ask users to explicitly pass trainable_variables to the decorated function.

Can you also edit the documentation? Can you add tests that show behavior which is broken without this explicit hatchet? Can you clarify what's the plan for code that creates variables when it's called?